src/query/execute_schema.c

```diff
@@ -10027,6 +10032,162 @@ do_alter_change_tbl_comment (PARSER_CONTEXT * const parser, PT_NODE * const alte
+  DB_CTMPL *ctemplate = NULL;
+  MOP class_mop = NULL;
+  OID class_oid;
+  bool tran_saved = false;
+
+  assert (alter_code == PT_CHANGE_COLUMN_COMMENT);
+
+  OID_SET_NULL (&class_oid);
+
+  entity_name = alter_node->info.alter.entity_name->info.name.original;
+  if (entity_name == NULL)
+    {
+      error = ER_UNEXPECTED;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, "Expecting a class or virtual class name.");
+      goto exit;
+    }
+
+  class_obj = db_find_class (entity_name);
+  if (class_obj == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+      goto exit;
+    }
+
+  error = locator_flush_class (class_obj);
+  if (error != NO_ERROR)
+    {
+      /* don't overwrite error */
+      goto exit;
+    }
+
+  /* force exclusive lock on class, even though it should have been already acquired */
+  if (locator_fetch_class (class_obj, DB_FETCH_WRITE) == NULL)
+    {
+      error = ER_FAILED;
+      goto exit;
+    }
+
+  ctemplate = dbt_edit_class (class_obj);
+  if (ctemplate == NULL)
+    {
+      /* when dbt_edit_class fails (e.g. because the server unilaterally aborts us), we must record the associated
+       * error message into the parser. Otherwise, we may get a confusing error msg of the form: "so_and_so is not a
+       * class". */
+      pt_record_error (parser, parser->statement_number - 1, alter_node->line_number, alter_node->column_number,
+		       er_msg (), NULL);
+      error = er_errid ();
+      goto exit;
+    }
+
+  error = tran_system_savepoint (UNIQUE_SAVEPOINT_CHANGE_COLUMN_COMMENT);
+  if (error != NO_ERROR)
+    {
+      goto exit;
+    }
+  tran_saved = true;
+
+  attr_node = alter_node->info.alter.alter_clause.attr_mthd.attr_def_list;
+  while (attr_node != NULL)
+    {
+      attr_name = get_attr_name (attr_node);
+
+      comment_node = attr_node->info.attr_def.comment;
+
+      assert (comment_node != NULL);
+      assert (comment_node->node_type == PT_VALUE);
+
+      meta = (attr_node->info.attr_def.attr_type == PT_META_ATTR);
+      shared = (attr_node->info.attr_def.attr_type == PT_SHARED);
+      name_space = (meta) ? ID_CLASS_ATTRIBUTE : ((shared) ? ID_SHARED_ATTRIBUTE : ID_ATTRIBUTE);
+
+      /* get the attribute structure */
+      error = smt_find_attribute (ctemplate, attr_name, (name_space == ID_CLASS_ATTRIBUTE) ? 1 : 0, &found_attr);
+      if (error != NO_ERROR)
+	{
+	  return error;
+	}
+
+      assert (found_attr != NULL);
+
+      /* comment */
+      comment_str = comment_node->info.value.data_value.str;
+
+      if (comment_str != NULL)
+	{
+	  found_attr->comment = ws_copy_string ((char *) pt_get_varchar_bytes (comment_str));
+	  if (found_attr->comment == NULL)
+	    {
+	      error = (er_errid () != NO_ERROR) ? er_errid () : ER_FAILED;
+	      goto exit;
+	    }
+	}
+      else if (comment_str == NULL)
+	{
+	  if (found_attr->comment != NULL)
+	    {
+	      ws_free_string (found_attr->comment);
+	      found_attr->comment = NULL;
+	    }
```

Comment:
```suggestion
	  ws_free_string_and_init (found_attr->comment);
```
NULL checking for found_attr->comment is going to be performed in ws_free_string

Reply:
After checking the "ws_free_string_and_init" macro and the "ws_free_string" function, it would be better to use the "ws_free_string_and_init" function as you said. Thank you.